e2e/cypress/integration/channel/channel_routing_spec.js

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Channel routing', () => {
+    before(() => {
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('should go to town square channel view', () => {
+        cy.visit('/ad-1/channels/town-square');
+        cy.get('#channelHeaderTitle').should('be.visible').should('contain', 'Town Square');
+    });
```

Comment:
are we testing that townsquare exists? I'm not really sure of the purpose of this test
if it is about routing, you should probably first go somewhere that is not town square?
also add conventions comments from lines 5-8 (see comment below)

Reply:
This test checks if the routing is working correctly. If the town square fails it would fail in the after all part but I want to add it as a new case to have it as documentation as well. No need to go to another channel and go back, if the routing is failing is going to fail with a 404 which is what a I want to check